pkg/provider/cloud/openstack/provider.go

```diff
@@ -164,6 +164,21 @@ func validateExistingSubnetOverlap(networkID string, netClient *gophercloud.Serv
 	})
 }
 
+func ensureFinalizers(cluster *kubermaticv1.Cluster, finalizers []string, update provider.ClusterUpdater) (*kubermaticv1.Cluster, error) {
```

Comment:
I started doing some small refactoring for the `InitializeCloudProvider` func. Without that lint would complain about cyclomatic complexity
```
pkg/provider/cloud/openstack/provider.go:169:1: cyclomatic complexity 31 of func `(*Provider).InitializeCloudProvider` is high (> 30) (gocyclo)
```